### PR TITLE
Additional functionality to enable workload generation

### PR DESCRIPTION
**Pull request by sjug**
Functions will pass CL config parameters to pods as environment variables on creation. Some functionality to get cluster state. Small HTTP function to support synchronized tests.

/cc @timothysc @gmarek @spiffxp @jayunit100 @jeremyeder

### CODE REVIEW

**Comment by timothysc** [comment]
before I dig too much into this, we really need some level of automation to ensure that 

- Everything compiles 
- The tests execute and give meaningful results  

Without this we are flying blind.

**Comment by timothysc** [comment]
So after fiddling + talking + more fiddling the consensus was to push the core functionality into libraries in that are in the main e2e testing repository.

**Comment by gmarek** [comment]
@sjug - can you fix the Travis (go get ... failures)

File: clusterloader/cluster_loader.go [hunk 1]
```diff
@@ -90,25 +100,17 @@ var _ = framework.KubeDescribe("Cluster Loader [Feature:ManualPerformance]", fun
 			gomega.Expect(err).NotTo(gomega.HaveOccurred())
 			framework.Logf("All pods running in namespace %s.", ns.Name)
 		}
+
+		// Start sync WS
```
Comment (gmarek) [inline clusterloader/cluster_loader.go]: Why this is commented out?

File: clusterloader/cluster_loader.go [hunk 2]
```diff
@@ -118,73 +120,7 @@ func mkPath(file string) string {
...
-
-	framework.Logf("The loaded config file is: %+v", configStruct.Spec.Containers)
-	return
-}
-
-func getNsCmdFlag(ns *v1.Namespace) string {
-	return fmt.Sprintf("--namespace=%v", ns.Name)
-}
-
+// appendIntToString appends an integer i to string s
```
Comment (gmarek) [inline clusterloader/cluster_loader.go]: you can just use `fmt.Sprintf("%s%d", string, int)`

Comment (sjug) [inline clusterloader/cluster_loader.go]: Nice, thanks!

File: clusterloader/cluster_loader.go (hunk 1 above)
Comment (sjug) [inline clusterloader/cluster_loader.go]: Not in use yet

File: clusterloader/config/rcpod.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+provider: local
```
Comment (gmarek) [inline clusterloader/config/rcpod.yaml]: When I see rc* I generally think "ReplicationController", so maybe change to consumerPod for sake of clarity.

File: clusterloader/content/rc-pod.json [hunk 4]
```diff
@@ -0,0 +1,54 @@
+{
+    "kind": "Pod",
+    "apiVersion": "v1",
+    "metadata": {
+        "name": "rc-pod",
```
Comment (gmarek) [inline clusterloader/content/rc-pod.json]: same as above - rc == ReplicationController

File: clusterloader/framework/util.go [hunk 5]
```diff
@@ -79,3 +106,248 @@ func CreatePods(f *framework.Framework, appName string, ns string, labels map[st
...
+				return
+			}
+		}
+		framework.Failf("No pod tuning found for: %s", podTuning)
+	}
+	return nil
+}
+
+// Server is the webservice that will syncronize the start and stop of Pods
+func Server(c *PodCount) error {
```
Comment (gmarek) [inline clusterloader/framework/util.go]: s/Server/StartServer/

File: clusterloader/framework/util.go [hunk 6]
```diff
@@ -79,3 +106,248 @@ func CreatePods(f *framework.Framework, appName string, ns string, labels map[st
...
+	fmt.Println("Listening on port", server.Addr)
+	select {
+	case <-c.Shutdown:
+		fmt.Println("Shutdown server")
+		ln.Close()
+		return err
+	}
+}
+
+func handleStart(fn http.HandlerFunc, c *PodCount) http.HandlerFunc {
```
Comment (gmarek) [inline clusterloader/framework/util.go]: startHandler

File: clusterloader/framework/util.go [hunk 7]
```diff
@@ -79,3 +106,248 @@ func CreatePods(f *framework.Framework, appName string, ns string, labels map[st
...
+	a[0] = unicode.ToLower(a[0])
+	return string(a)
+}
+
+// convertVariables takes our loaded struct and converts it into a map[string]string.
+func convertVariables(params ParameterConfigType) map[string]string {
+	m := structs.Map(params)
+	values := make(map[string]string)
+	for k, v := range m {
+		k = firstLowercase(k)
```
Comment (gmarek) [inline clusterloader/framework/util.go]: Why this is needed? If names are case sensitive we should use this, if they're not just cast whole key.

File: clusterloader/framework/util.go [hunk 8]
```diff
@@ -79,3 +106,248 @@ func CreatePods(f *framework.Framework, appName string, ns string, labels map[st
...
+	freshConfigVars := convertVariables(vars)
+	dirtyConfigVars := getClusterData(f, freshConfigVars)
+	configMap := newConfigMap(ns, configMapName, dirtyConfigVars)
+	framework.Logf("Creating configMap %v in namespace %v", configMap.Name, ns)
+	var err error
+	if configMap, err = f.ClientSet.Core().ConfigMaps(ns).Create(configMap); err != nil {
+		framework.Failf("Unable to create test configMap %s: %v", configMap.Name, err)
+	}
+
+	for i, envVar := range config.Spec.Containers[0].Env {
```
Comment (gmarek) [inline clusterloader/framework/util.go]: I don't know how this will work. My gut feeling is that kubelet should handle this. @yujuhong

**Comment by gmarek** [comment]
some minor comments. Sorry it took that long, I finally have some cycles to use on something that's not forgiveness...

**Comment by gmarek** [comment]
(sorry - missclick)

**Comment by sjug** [comment]
No problem @gmarek, I've got to work on some pending deadlines here so I haven't had as much time to work on this functionality. I appreciate the review.

**Comment by gmarek** [comment]
@sjug - is this PR live or should we close it?

**Comment by sjug** [comment]
@gmarek I need to rework this significantly given it's so old. Lets close and I can open another one when it is ready.

**Pull request closed without merging**